suzieq/engines/pandas/routes.py

```diff
@@ -49,8 +44,13 @@ def get(self, **kwargs):
         addnl_fields = kwargs.pop('addnl_fields', [])
         user_query = kwargs.pop('query_str', '')
 
-        columns = kwargs.get('columns', ['default'])
-        addnl_fields, drop_cols = self._cons_addnl_fields(
+        columns = kwargs.pop('columns', ['default'])
+        fields = self.schema.get_display_fields(columns)
+        if addnl_fields:
+            # if user requests additional fields, honor returning them
+            fields += [x for x in addnl_fields if x not in fields]
```

Comment:
Aren't we using `addnl_fields` only internally? We should only return the columns specified with `columns`.
Otherwise what is the purpose of `addnl_fields`?

Reply:
Sometimes more complex tables such as topology and path uses them, and they need them. Why? Because its easier to say default + additional fields. We'll add the "+" and "-" notation later so that we can do things like columns='+macaddr' meaning the default columns plus macaddr, later. At that point, this remark will be true.

Reply:
We don't to that for most of the tables. Have a look at Device, Bgp, EvpnVni just to name the first that I checked. We always return `df[fields]` where fields is not extended with `addnl_fields`.

Reply:
Those tables are not called by the other more complex tables.

Reply:
OK, routes is the sole one who does this @claudious96 , and so I'll fix it correctly and remove this.

Reply:
Interfaces does the same thing as routes, and both are invoked by more complex tables such as network, topology and path. One reason only interfaces and routes do it is because in the other cases, either the default list is enough or the list of fields required is small enough that we can specify the set.